platforms/software/plugins-distribution/src/main/java/org/gradle/api/distribution/plugins/DistributionBasePlugin.java

```diff
@@ -147,7 +147,7 @@ private static <T extends AbstractArchiveTask> TaskProvider<T> addArchiveTask(
             CopySpec childSpec = project.copySpec();
             childSpec.with(distribution.getContents());
             childSpec.into((Callable<String>) () ->
-                TextUtil.minus(task.getArchiveFileName().get(), "." + task.getArchiveExtension().get())
+                TextUtil.removeLast(task.getArchiveFileName().get(), "." + task.getArchiveExtension().get())
```

Comment:
If adding a new method to `TextUtil` is not the preferred approach, we can replace this
```suggestion
                StringUtils.removeEnd(task.getArchiveFileName().get(), "." + task.getArchiveExtension().get())
```

Reply:
I don't feel strongly about this either way.  I think adding something to `TextUtil` is fine.

Reply:
Thank you. I will maintain the current state.